### PR TITLE
Gas Optimizations

### LINKED ISSUE #4404: contracts/utils/math: replace "if (A && B)" to save gas fees
The solidity compiler generates more code for “if (A && B) {}” than for “if (A) {if (B) {} }”.
Our experiment shows that switching from “if (A && B) {}” to “if (A) {if (B) {} }” can save gas fee.

**💻 Environment**

master branch

**📝 Details**

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/874c2d3c02ec1bce6af9a30bc828d3fe2079136b/contracts/utils/math/Math.sol#L217-L223

Replacing the following code snippet
```
        if (rounding == Rounding.Up && mulmod(x, y, denominator) > 0) {
            result += 1;
        }
```
with
```
       if (rounding == Rounding.Up) {
            if (mulmod(x, y, denominator) > 0) {
                result += 1;
            }
       }
```

can save gas fees and does not hurt code readability.

**🔢 Code to reproduce bug**

### PR DESCRIPTION
**Pull request by SensationalShubham**
Fixes #4404  

The `&&` condition in If-else loop costs more than if it breaks in a nested loop
Eg. 
         It costs more gas
```
if(a>0 && b>5){}
```   
  then
```
if(a>0){
    if(b>5){}
   
 }
```
so, I replace `&&` conditions with their nested loop all over the Repo.

#### PR Checklist

- [ ] Tests
- [✅ ] Documentation
- [✅] Changeset entry (run `npx changeset add`)

### CODE REVIEW

**Review by SensationalShubham — COMMENTED** [review]
Trying to add Changeset

**Review by SensationalShubham — COMMENTED** [review]
Finally added changeset

File: contracts/governance/Governor.sol [hunk 1]
```diff
@@ -191,8 +191,10 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor, IERC721Receive
             return ProposalState.Active;
         }
 
-        if (_quorumReached(proposalId) && _voteSucceeded(proposalId)) {
-            return ProposalState.Succeeded;
+        if (_quorumReached(proposalId)) {
+            if (_voteSucceeded(proposalId)) {
```
Comment (frangio) [inline contracts/governance/Governor.sol:195]: These things are not equivalent! Look at the else branch.

**Comment by frangio** [comment]
Closing to discuss in issue first. Please wait until maintainers have agreed to a suggestion before submitting a PR.

**Pull request closed without merging**